### PR TITLE
Use editability conditions in build property pages

### LINKED ISSUE #8204: Allow properties to define an expression to control when they are editable
Today we have an expression that controls property visibility.

We could add a similar expression that controls when a property is marked as read only, conditional upon other property values.

This facility would simplify the approach taken in #8196 for example. In that PR, we add a duplicate of every property that might potentially be disabled, using mutually exclusive visibility conditions to hide the editable version and show the readonly one, based on other property values.

Here's an example of how this might work. The `EditabilityCondition` allows the UI to toggle the state of the UI element dynamically at runtime.

```xml
  <DynamicEnumProperty Name="SomeProperty"
                       ...>
    <DynamicEnumProperty.Metadata>
      <NameValuePair Name="EditabilityCondition">
        <NameValuePair.Value>
          (not (has-evaluated-value "Build" "WarningSeverity" "DisableAll"))
        </NameValuePair.Value>
      </NameValuePair>
    </DynamicEnumProperty.Metadata>
  </DynamicEnumProperty>
```

The majority of properties would not define a `EditabilityCondition`, and there would be no overhead for those properties.

### PR DESCRIPTION
**Pull request by adamint**
Editability conditions replace the need for readonly properties, so we can remove the existing readonly properties in build property pages. This is a draft PR while the impl of the EditabilityCondition is done.

Resolves #8204 

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/project-system/pull/8291)

### CODE REVIEW

**Review by drewnoakes — APPROVED** [review]

**Comment by adamint** [comment]
This can be merged once https://devdiv.visualstudio.com/DevDiv/_git/CPS/pullrequest/410209 is in VS main. 

Additionally, https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1566265/ should be set to resolved after this PR is merged.

**Review by MiYanni — APPROVED** [review]

File: docs/repo/property-pages/property-specification.md
Comment (tmeschter) [inline docs/repo/property-pages/property-specification.md:245]: You could add a link to that .xaml file in source.

File: docs/repo/property-pages/visibility-conditions.md
Comment (tmeschter) [inline docs/repo/property-pages/visibility-conditions.md:13]: Err, "four kinds" not "three four".

File: docs/repo/property-pages/visibility-conditions.md
Comment (tmeschter) [inline docs/repo/property-pages/visibility-conditions.md:15]: I wouldn't call this a visibility condition per se, as it doesn't affect the visibility of anything. :-)

**Review by tmeschter — APPROVED** [review]

File: docs/repo/property-pages/visibility-conditions.md
Comment (drewnoakes) [reply docs/repo/property-pages/visibility-conditions.md:13]: Also singular _condition_.

File: docs/repo/property-pages/visibility-conditions.md
Comment (drewnoakes) [reply docs/repo/property-pages/visibility-conditions.md:15]: ...should _be_ editable...

File: docs/repo/property-pages/visibility-conditions.md
Comment (adamint) [reply docs/repo/property-pages/visibility-conditions.md:13]: as seen by the sentences I wrote, I can english very well @tmeschter 😄 
@drewnoakes plural conditions is correct in american english

**Comment by adamint** [comment]
Note: I updated the docs to refer to the broader category of visibility+isreadonly conditions as "property conditions." If there's something I should be calling them instead, please let me know @drewnoakes

File: docs/repo/property-pages/visibility-conditions.md
Comment (drewnoakes) [reply docs/repo/property-pages/visibility-conditions.md]: Thanks. TIL there is a difference of opinion on this! https://english.stackexchange.com/questions/5539/types-of-things-vs-types-of-thing

I'd write:
- Four visibility conditions
- Four kinds of visibility condition

Apparently Edge would write:

image

😆

**Review by drewnoakes — APPROVED** [review]

**Pull request merged**
